### PR TITLE
CLDSRV-462 Expiration header is not compatible with legacy object md

### PR DESCRIPTION
**Pull request by nicolas2bert**
Before the Object Metadata refactor done around May 31, 2017 (https://github.com/scality/Cloudserver/commit/c22e44f63d571a6271af1f1a9c03266b9fd273ae), if no tags were set, the object tag was stored as undefined.

After the commit, if no tags are set, the object tag is stored as an empty object '{}'.

When the expiration response headers were implemented on https://github.com/scality/cloudserver/commit/812b09afeff5313ae3ec5e0adf99060b94c79b0d around Nov 22, 2021, the empty object was handled, but not the undefined tag logic, which made the expiration response headers not backward compatible.

We need to address both cases: the undefined property and the empty object '{}'.

### CODE REVIEW

**Comment by nicolas2bert** [comment]
ping

**Comment by nicolas2bert** [comment]
@bert-e reset

**Review by benzekrimaha — APPROVED** [review]
LGTM

File: lib/api/apiUtils/object/expirationHeaders.js [hunk 1]
```diff
@@ -37,8 +37,10 @@ const AMZ_ABORT_ID_HEADER = 'x-amz-abort-rule-id';
 
 function _generateExpHeadersObjects(rules, params, datetime) {
     const tags = {
-        TagSet: Object.keys(params.tags)
-            .map(key => ({ Key: key, Value: params.tags[key] })),
+        TagSet: params.tags
+            ? Object.keys(params.tags)
+                .map(key => ({ Key: key, Value: params.tags[key] }))
```
Comment (jonathan-gramain) [inline lib/api/apiUtils/object/expirationHeaders.js:42]: Small enhancement suggestion, not related to this PR's change (feel free to ignore):

```suggestion
            ? Object.entries(params.tags)
                .map(([key, value]) => ({ Key: key, Value: value }))
```

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "s3",
-  "version": "7.10.31",
+  "version": "7.10.32",
```
Comment (jonathan-gramain) [inline package.json:3]: Can you do a separate commit for the version bump? It helps when it comes to backporting things (as well as keeping track of version changes more easily).

Comment (nicolas2bert) [reply package.json:3]: Done, I agree. I usually split it.

**Comment by nicolas2bert** [comment]
@bert-e approve

**Comment by nicolas2bert** [comment]
@bert-e reset

**Comment by nicolas2bert** [comment]
@bert-e reset

**Review by jonathan-gramain — APPROVED** [review]

**Comment by nicolas2bert** [comment]
@bert-e approve
